crates/hir_def/src/generics.rs

```diff
@@ -73,9 +73,9 @@ pub enum WherePredicateTypeTarget {
 }
 
 #[derive(Default)]
-pub struct SourceMaps {
-    pub type_params: ArenaMap<LocalTypeParamId, Either<ast::Trait, ast::TypeParam>>,
-    pub lifetime_params: ArenaMap<LocalLifetimeParamId, ast::LifetimeParam>,
+pub(crate) struct SourceMaps {
```

Comment:
```suggestion
pub(crate) struct SourceMap {
```

a source map can be heterogeneous (see BodySourceMap)